src/react/hooks.js

```diff
@@ -48,3 +52,298 @@ export const useTrackColorSchemeChoice = () => {
+     * based on the current theme variant.
+     */
+    const setThemeVariantLoaded = (themeVariant) => {
+      if (themeVariant === 'light') {
+        onLoadThemeVariantLight();
+      }
+    };
+
+    /**
+     * Iterate over each theme variant URL and inject it into the HTML document, if it doesn't already exist.
+     */
+    Object.entries(themeVariants).forEach(([themeVariant, value]) => {
+      // If there is no config for the theme variant URL, set the theme variant to loaded and continue.
+      if (!value.url) {
+        setThemeVariantLoaded(themeVariant);
+        return;
+      }
+      let themeVariantLink = document.head.querySelector(`link[href='${value.url}']`);
+      const stylesheetRelForVariant = generateStylesheetRelAttr(themeVariant);
+      if (!themeVariantLink) {
+        const getExistingThemeVariantLinks = () => document.head.querySelectorAll(`link[data-paragon-theme-variant='${themeVariant}']`);
+        // find existing links
+        const existingLinks = getExistingThemeVariantLinks();
+
+        // create new link
+        const createThemeVariantLink = (url) => {
+          themeVariantLink = document.createElement('link');
+          themeVariantLink.href = url;
+          themeVariantLink.rel = 'stylesheet';
+          themeVariantLink.dataset.paragonThemeVariant = themeVariant;
+          themeVariantLink.onload = () => {
+            setThemeVariantLoaded(themeVariant);
+            existingLinks.forEach((link) => {
+              link.remove();
+            });
+          };
+          themeVariantLink.onerror = () => {
+            logError(`Failed to load theme variant (${themeVariant}) CSS from ${value.url}`);
+            if (PARAGON?.themeUrls?.variants?.[themeVariant]) {
+              const themeVariantFallbackUrl = `${getConfig().BASE_URL}/${PARAGON.themeUrls.variants[themeVariant]}`;
+              logInfo(`Falling back to locally installed theme variant (${themeVariant}) CSS: ${themeVariantFallbackUrl}`);
+              themeVariantLink = createThemeVariantLink(themeVariantFallbackUrl);
+              const otherExistingLinks = getExistingThemeVariantLinks();
+              otherExistingLinks.forEach((link) => {
+                link.remove();
+              });
+              document.head.insertAdjacentElement(
+                'afterbegin',
+                themeVariantLink,
+              );
+            }
+          };
+          return themeVariantLink;
+        };
+        themeVariantLink = createThemeVariantLink(value.url);
+        document.head.insertAdjacentElement(
+          'afterbegin',
+          themeVariantLink,
+        );
+      } else if (themeVariantLink.rel !== stylesheetRelForVariant) {
+        themeVariantLink.rel = stylesheetRelForVariant;
+      }
+    });
+  }, [themeVariants, currentThemeVariant, onLoadThemeVariantLight]);
+};
+
+const handleParagonVersionSubstitution = (url) => {
+  if (!url || !url.includes('$paragonVersion') || !PARAGON?.version) {
```

Comment:
Alright, I'll bite.  I can't for the life of me figure out where `PARAGON` is coming from 😄

Reply:
Ha, magic no more: https://github.com/openedx/frontend-build/pull/365